src/render_threaded_fts_bonus.c

```diff
@@ -0,0 +1,115 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   threads_alt_versions_bonus.c                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/01/14 10:14:59 by gwolf             #+#    #+#             */
+/*   Updated: 2024/01/22 10:44:53 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "render_bonus.h"
+
+void	*ft_raytrace_sample_threaded(void *arg)
+{
+	t_render		*render;
+	t_vec2i			pos;
+	t_ray			ray;
+	t_colour		colour;
+
+	render = ((t_thread_data *)arg)->arg;
+	ray.origin = render->cam.centre;
+	ray.d = 1.0;
+	pos.y = ((t_thread_data *)arg)->id;
+	while (pos.y < render->cam.image.y)
+	{
+		pos.x = -1;
+		while (++pos.x < render->cam.image.x)
+		{
+			colour = ft_shoot_ray(pos, ray, render->scene, render->cam);
+			render->sample_buffer[pos.y * render->cam.image.x + pos.x] = colour;
+		}
+		pos.y += NUM_THREADS;
+	}
+	return (NULL);
+}
+
+void	*ft_add_raytrace_sample_threaded(void *arg)
+{
+	t_render	*render;
+	t_vec2i		pos;
+	t_ray		ray;
+	t_colour	colour;
+
+	render = ((t_thread_data *)arg)->arg;
+	ray.origin = render->cam.centre;
+	ray.d = 1.0;
+	pos.y = ((t_thread_data *)arg)->id;
+	while (pos.y < render->cam.image.y)
+	{
+		pos.x = -1;
+		while (++pos.x < render->cam.image.x)
+		{
+			colour = ft_shoot_aa_ray(pos, ray, render->scene, render->cam);
+			colour = ft_add_colour(colour,
```

Comment:
`render->[...] = ft_add_colour(`
This line is already too long. Would need to rename variable sample_buffer to only buffer or something

Reply:
Yeah, I meant you dont need to assign it to colour and then assign colour to render->[...]

Reply:
`render->sample_buffer[pos.y * render->cam.image.x + pos.x] = ft_add_colour(`
is too long.

Pushing the function to the next line also does not work, because then the last line is too long
```
render->sample_buffer[pos.y * render->cam.image.x + pos.x] =
   ft_add_colour(colour,
         render->sample_buffer[pos.y * render->cam.image.x + pos.x]);;
```

Or am I missing something?

Reply:
You could line break again after the asteriks *

Reply:
Which would give us 4 lines instead of 3..

Performance wise it makes no difference, as the compiler will optimize away (even if not, would be in the space of microseconds).
Readability wise it gets more confusing as one assignment spans over 4 lines.

If you really don't like it I can rename the sample_buffer to buffer but I don't like the 4 line assignment.

Reply:
naaa, I meant like
```
render->sample_buffer[pos.y * render->cam.image.x + pos.x] =
   ft_add_colour(colour, render->sample_buffer[pos.y * \
   render->cam.image.x + pos.x]);
```
It's not for performance, but it's a hiccup to see the consecutive assignment of colour = sth.1 and the next line sth.2 = colour instead of sth2 = sth1

Reply:
```
colour = ft_shoot_aa_ray(pos, ray, render->scene, render->cam);
render->sample_buffer[pos.y * render->cam.image.x + pos.x]
	= ft_add_colour(colour, render->sample_buffer[pos.y * \
		render->cam.image.x + pos.x]);
```
v
```
colour = ft_shoot_aa_ray(pos, ray, render->scene, render->cam);
colour = ft_add_colour(colour,
		render->sample_buffer[pos.y * render->cam.image.x + pos.x]);
render->sample_buffer[pos.y * render->cam.image.x + pos.x] = colour;
```

I don't get it. Changed.